lib/Server.js

```diff
@@ -382,12 +382,23 @@ class Server {
               }
 
               const bypass = typeof proxyConfig.bypass === 'function';
```

Comment:
Setting the default value of proxyConfig.bypass to a no-op function would dramatically simplify the code here (assuming noop default):

```suggestion
              const bypassUrl = proxyConfig.bypass(req, res, proxyConfig);
              
              if (bypassUrl) req.url = bypassUrl;
              else if (bypassUrl === false) req.url = null;
              
              next();
```

It would also help if the flag for `bypass` to return was `null` (since that's apparently what req.url wants), instead of mixing `false` and `null` (but that would probably be a breaking change). I think this is right, but the suggested logic is nigh impossible to follow after re-reading it for 15 minutes:

```js
              const bypassUrl = proxyConfig.bypass(req, res, proxyConfig);
              
              if (bypassUrl || bypassUrl === null) req.url = bypassUrl;
              
              next();
```